packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -715,6 +716,7 @@ abstract class ResidentRunner {
   /// Throws an [AssertionError] if [Devce.supportsScreenshot] is not true.
   Future<void> screenshot(FlutterDevice device) async {
     assert(device.device.supportsScreenshot);
+
     final Status status = logger.startProgress('Taking screenshot for ${device.device.name}...', timeout: timeoutConfiguration.fastOperation);
```

Comment:
Not necessarily for this PR: The status.cancel()s for this status should go in a finally block for the outermost try instead of spamming them before every return.

Reply:
I'll roll this into my cleanup PR